### PR TITLE
feat: adding /about command as before

### LINKED ISSUE #6: Command about necessary information about the bot
`/about` command which will display information links about dashboard, account settings, premium, privacy policy

Description about status, top.gg (voting), support server. All of these would be links

Short description about the bot itself 

Footer will display the response time & shard the server is on

### PR DESCRIPTION
**Pull request by CringleySDays**
- Added `/about` command as it was in v7 without any changes

**Notice:** @ConnysCode can you run this code in the host server (using a testing bot) to see if the response time is sent accurately. This depends if the clock is the same as the discord clock. Otherwise it'd return negative results in the 1000s before merging please.

- Error handling is left for future when a proper logging system is in place

### CODE REVIEW

**Comment by ConnysCode** [comment]
@CringleySDays As long as the clock is set correctly, the timezone should not matter. The timestamp is just the time in seconds/milliseconds passed since epoch. This is the same for every timezone.

File: src/commands/info/about.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { ActionRowBuilder, ButtonBuilder, ButtonStyle, ChatInputCommandInteraction, SlashCommandBuilder } from 'discord.js';
+import defaultEmbed from '../../helpers/messaging/default-embed';
+import Command from '../../classes/command';
+
+const execute = async (interaction: ChatInputCommandInteraction) => {
+  if (!interaction.guildId || !interaction.user.id || !interaction.guild) return;
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: The command will just time out. Please add a TODO, to hint that the logging system is still missing there.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

**Comment by ConnysCode** [comment]
@CringleySDays Please pull the recent development branch and push that again. <3

**Review by ConnysCode — APPROVED** [review]

**Comment by ConnysCode** [comment]
Also please add localization. Refer to https://github.com/xp-bot/xp-bot/blob/3d6da0bc362abcc09c5f84104843279110827086/CONTRIBUTING.md?plain=1#L33-L61

Check out the level command for reference:
https://github.com/xp-bot/xp-bot/blob/3d6da0bc362abcc09c5f84104843279110827086/src/commands/info/level.ts#L38-L44

**Comment by CringleySDays** [comment]
Sure, I'll do that once I'm home

**Comment by CringleySDays** [comment]
I'm not sure if this is how you want me to do it, I think I did it. I'm not too sure sorry

File: src/commands/info/about.ts [hunk 2]
```diff
@@ -1,45 +1,58 @@
...
-  const aboutInfoEmbed = defaultEmbed('normal')
+  const aboutInfoEmbed = defaultEmbed(DefaultEmbedType.NORMAL)
     .setTitle('Reimagine your Community')
     .addFields(
       {
         name: 'Top-Tier Leveling Solution',
-        value: 'Elevate your Community to the next Level with Top-Tier Leveling, endless Customisability and more.',
-        inline: false
+        value:
+          'Elevate your Community to the next Level with Top-Tier Leveling, endless Customisability and more.',
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: Every string needs to be localised.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

**Comment by CringleySDays** [comment]
The way I did it for footer is correct right?

File: src/commands/info/about.ts [hunk 3]
```diff
@@ -25,24 +25,58 @@ const execute = async (interaction: ChatInputCommandInteraction) => {
     .setTitle('Reimagine your Community')
     .addFields(
       {
-        name: 'Top-Tier Leveling Solution',
-        value:
-          'Elevate your Community to the next Level with Top-Tier Leveling, endless Customisability and more.',
+        name: t('fields.summary_title', {
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: Where are these localisations supposed to come from? You gotta define them in the `/locales/{{lng}}/{{namespace}}.json` files.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

File: src/locales/en/about_command.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "command_info": {
+    "description": "Get information about XP.",
+    "name": "about"
+  },
+  "title": {
+    "title": "Reimagine your Community"
+  },
+  "fields": {
```
Comment (ConnysCode) [inline src/locales/en/about_command.json]: categories are meant to be in a singular wording.

```diff
+ field
- fields
```

File: src/locales/en/about_command.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "summary_title": "Top-Tier Leveling Solution",
+    "summary_description": "Elevate your Community to the next Level with Top-Tier Leveling, endless Customisability and more.",
+
+    "server_support_title": "Join our Server",
+    "fields.server_support_link": "[Join XP]({{link}})",
```
Comment (ConnysCode) [inline src/locales/en/about_command.json]: duplicated category "fields" in key.

File: src/locales/en/about_command.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+    "vote_title": "Vote for XP!",
+    "vote_link": "[top.gg]({{link}})",
+
+    "status_title": "Service Status",
+    "status_link": "[Status]({{link}})"
+  },
+  "footer": {
+    "response_shard_company_info": "Response time: {{response}}ms • Shard: {{shard}} • © {{company}}"
```
Comment (ConnysCode) [inline src/locales/en/about_command.json]: These symbols are a headache to translate.
Please split them individual keys.
F.e. `response_time_n`, `shard_n`, ...

File: src/commands/info/about.ts [hunk 7]
```diff
@@ -22,7 +22,7 @@ const execute = async (interaction: ChatInputCommandInteraction) => {
   }
 
   const aboutInfoEmbed = defaultEmbed(DefaultEmbedType.NORMAL)
-    .setTitle('Reimagine your Community')
+    .setTitle(`${t('title.title', { lng: 'en', ns: 'about_command' })}}`)
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: why is the title in the title category?

just key `title`, no category needed.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

**Comment by CringleySDays** [comment]
I removed the symbols and replaced it with commas

**Comment by ConnysCode** [comment]
> I removed the symbols and replaced it with commas

No, the about command is to be the exact same as it is now. The symbols need to stay, just not as part of the translation string.

File: src/commands/info/about.ts [hunk 8]
```diff
@@ -22,28 +22,28 @@ const execute = async (interaction: ChatInputCommandInteraction) => {
...
       {
-        name: t('fields.server_title', {
+        name: t('field.server_title', {
           //'Join our Server'
           lng: 'en',
           ns: 'about_command',
         }),
-        value: t('fields.server_link', {
+        value: t('field.server_link', {
           //'[Join XP](https://discord.com/invite/ccTAnzw)'
```
Comment (ConnysCode) [inline src/commands/info/about.ts:45]: The links should be part of the translation! Only the "Join XP" should be translatable. Why would we translate a discord.com link?

File: src/locales/en/about_command.json [hunk 9]
```diff
@@ -20,6 +18,8 @@
     "status_link": "[Status]({{link}})"
   },
   "footer": {
-    "response_shard_company_info": "Response time: {{response}}ms • Shard: {{shard}} • © {{company}}"
+    "response_time_n": "Response time: {{response}}ms",
+    "shard_n": "Shard: {{shard}}",
+    "company_n": "by {{company}}"
```
Comment (ConnysCode) [inline src/locales/en/about_command.json]: `by_company`.

`n` is a placeholder hinting to a number. Strings are to be expected to be interpolated. Numbers potentially follow different logics.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

File: src/locales/en/about_command.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{
+  "command_info": {
+    "description": "Get information about XP.",
+    "name": "about"
+  },
+  "title": {
+    "title": "Reimagine your Community"
```
Comment (ConnysCode) [inline src/locales/en/about_command.json]: Now that i think of it, this should also not be translated, due to this being our slogan. Slogans do not get translated.

**Comment by CringleySDays** [comment]
I think I did it as you told me to, I'm not too sure 😅

File: src/commands/info/about.ts (hunk 8 above)
Comment (ConnysCode) [reply src/commands/info/about.ts:45]: Idk but this is still not correct.

Only "Join XP" should be localized.

File: src/commands/info/about.ts [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+        value: '[top.gg](https://top.gg/bot/706935674800177193)',
+        inline: true,
+      },
+      {
+        name: t('field.status_title', {
+          //'Service Status'
+          lng: 'en',
+          ns: 'about_command',
+        }),
+        value: t('field.status_link', {
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: Here as well, only "Status" should be localized.

File: src/commands/info/about.ts [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+            ns: 'about_command',
+            company: 'namespace.media',
+          })}`,
+        }),
+      ],
+
+      components: [aboutActionRow, aboutActionRow2],
+    });
+  } catch (error) {
+    console.error(error);
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: BTW error system has been implemented. Please throw a "new XPError" wherever code could fail.

File: src/commands/info/about.ts [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+          //'[Status](https://xp-bot.net/status)'
+          lng: 'en',
+          ns: 'about_command',
+        }),
+        inline: true,
+      },
+    );
+
+  const dashboardButton = new ButtonBuilder()
+    .setLabel('Server Dashboard')
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: These buttons also have not been localized.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

File: src/commands/info/about.ts (hunk 8 above)
Comment (ConnysCode) [reply src/commands/info/about.ts:45]: ??

Comment (ConnysCode) [reply src/commands/info/about.ts:45]:
```ts
value: `[${t('field.server_link'), {...}}](https://discord.com/invite/ccTAnzw)`
```

**Review by ConnysCode — CHANGES_REQUESTED** [review]

File: src/commands/info/about.ts [hunk 14]
```diff
@@ -10,15 +10,15 @@ import defaultEmbed, {
   DefaultEmbedType,
 } from '../../helpers/messaging/default-embed';
 import { t } from 'i18next';
-import XPError from '../../classes/xp-error';
+import XPError, { XPErrorType } from '../../classes/xp-error';
 
 const execute = async (interaction: ChatInputCommandInteraction) => {
   if (!interaction.guildId || !interaction.user.id || !interaction.guild) {
     await interaction.reply({
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: Don't reply when throwing an XPError. The error handler does that for you.

File: src/commands/info/about.ts [hunk 15]
```diff
@@ -10,15 +10,15 @@ import defaultEmbed, {
...
       content: 'Invalid interaction.',
       ephemeral: true,
     });
-    throw new XPError('Failed to send /about embed');
+    throw new XPError(XPErrorType.INTERACTION_GUILD_UNRESOLVABLE);
```
Comment (ConnysCode) [inline src/commands/info/about.ts:17]: This error is only for not being able to resolve a guild. Please add a separate condition for not being able to resolve the user id.

File: src/commands/info/about.ts [hunk 16]
```diff
@@ -155,7 +155,7 @@ const execute = async (interaction: ChatInputCommandInteraction) => {
       components: [aboutActionRow, aboutActionRow2],
     });
   } catch (error) {
-    throw new XPError('Failed to send /about embed with buttons in line 136');
+    throw new XPError(XPErrorType.INTERACTION_GUILD_UNRESOLVABLE);
```
Comment (ConnysCode) [inline src/commands/info/about.ts]: But thats not what's happening here.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

File: src/commands/info/about.ts [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+} from 'discord.js';
+import Command from '../../classes/command';
+import defaultEmbed, {
+  DefaultEmbedType,
+} from '../../helpers/messaging/default-embed';
+import { t } from 'i18next';
+import XPError, { XPErrorType } from '../../classes/xp-error';
+
+const execute = async (interaction: ChatInputCommandInteraction) => {
+  if (!interaction.guildId || !interaction.guild) {
```
Comment (ConnysCode) [inline src/commands/info/about.ts:16]: We don't use the interaction.guild, so we don't have to check for it.

File: src/commands/info/about.ts [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+    throw new XPError(XPErrorType.INTERACTION_GUILD_UNRESOLVABLE);
+  } else if (!interaction.user.id) {
+    throw new XPError(XPErrorType.INTERACTION_USER_UNRESOLVABLE);
+  }
+
+  const aboutInfoEmbed = defaultEmbed(DefaultEmbedType.NORMAL)
+    .setTitle('Reimagine your Community')
+    .addFields(
+      {
+        name: t('field.summary_title', {
```
Comment (ConnysCode) [inline src/commands/info/about.ts:26]: `field.summary.title`

File: src/commands/info/about.ts [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+          //'Top-Tier Leveling Solution'
+          lng: 'en',
+          ns: 'about_command',
+        }),
+        value: t('field.summary_description', {
```
Comment (ConnysCode) [inline src/commands/info/about.ts:31]: `field.summary.description`

File: src/commands/info/about.ts [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+          //'Elevate your Community to the next Level with Top-Tier Leveling, endless Customisability and more.'
+          lng: 'en',
+          ns: 'about_command',
+        }),
+        inline: false,
+      },
+      {
+        name: t('field.server_title', {
```
Comment (ConnysCode) [inline src/commands/info/about.ts:39]: See comments above.

File: src/commands/info/about.ts [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+      embeds: [
+        aboutInfoEmbed.setFooter({
+          text: `${t('footer.response_time_n', {
+            lng: 'en',
+            ns: 'about_command',
+            response: Date.now() - interaction.createdTimestamp,
+          })} • ${t('footer.shard_n', {
+            lng: 'en',
+            ns: 'about_command',
+            shard: interaction.guild.shardId + 1,
```
Comment (ConnysCode) [inline src/commands/info/about.ts:144]: `discordClient.shard?.ids[0]`

File: src/commands/info/about.ts [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+            ns: 'about_command',
+            company: 'namespace.media',
+          })}`,
+        }),
+      ],
+
+      components: [aboutActionRow, aboutActionRow2],
+    });
+  } catch (error) {
+    throw new XPError(XPErrorType.INTERACTION_USER_UNRESOLVABLE);
```
Comment (ConnysCode) [inline src/commands/info/about.ts:156]: how is that an accurate description for what failed here?

File: yarn.lock [hunk 23]
```diff
@@ -7,6 +7,8 @@
   resolved "https://registry.yarnpkg.com/@aashutoshrathi/word-wrap/-/word-wrap-1.2.6.tgz#bd9154aec9983f77b3a034ecaa015c2e4201f6cf"
   integrity sha512-1Yjs2SvM8TflER/OD3cOjhWWOZb58A2t7wpE2S9XfBYTiIl+XFhQG2bjy4Pu1I+EAlCNUzRDYDdFwFYUKvXcIA==
 
+<<<<<<< HEAD
```
Comment (ConnysCode) [inline yarn.lock:10]: Merge annotations.

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -57,6 +57,10 @@ Key categories will always be named in a singular form. (e.g. button.x instead o
 
 - Every command has to have a namespace. The namespace is the command name in snake_case followed by `_command`. (e.g. set_xp_command, party_command, ...)
 - Every command's namespace, has to have the following keys:
+<<<<<<< HEAD
```
Comment (ConnysCode) [inline CONTRIBUTING.md:60]: Merge annotations.

**Review by ConnysCode — CHANGES_REQUESTED** [review]

**Pull request closed without merging**